### PR TITLE
added alpha2 release notes

### PR DESCRIPTION
**Pull request by evilnick**

### CODE REVIEW

File: en/release-notes.md [hunk 1]
```diff
@@ -25,6 +25,96 @@ The official Python client library for MAAS is available in the Ubuntu 18.04
 LTS package archive or you can download the source from:
 [https://github.com/maas/python-libmaas/releases](https://github.com/maas/python-libmaas/releases)
 
+## 2.4.0 (alpha2)
+
+### NTP services now provided by Chrony
+
+Starting with 2.4 Alpha 2, and in common with changes being made to Ubuntu Server, MAAS replaces ‘ntpd’ with Chrony for the NTP protocol. MAAS will handle the upgrade process and automatically resume NTP service operation.
```
Comment (caldav) [inline en/release-notes.md]: `ntpd` maybe?

File: en/release-notes.md [hunk 2]
```diff
@@ -25,6 +25,96 @@ The official Python client library for MAAS is available in the Ubuntu 18.04
...
+
+### Vanilla CSS Framework Transition
+
+MAAS 2.4 is undergoing a Vanilla CSS framework transition to a new version of vanilla, which will bring a fresher look to the MAAS UI. This framework transition is currently work in progress and not all of the UI have been fully updated. Please expect to see some inconsistencies in this new release.
```
Comment (caldav) [inline en/release-notes.md]: "new version of Vanilla,"
"a work in progress"

File: en/release-notes.md [hunk 3]
```diff
@@ -25,6 +25,96 @@ The official Python client library for MAAS is available in the Ubuntu 18.04
...
+MAAS 2.4 is currently undergoing major surgery to improve various areas of operation
+that are not visible to the user. These updates will improve the overall performance of
+MAAS in larger environments. These improvements include:
+
+
+ - **AsyncIO based event loop**
+    MAAS has an event loop which performs various internal actions. In older versions of MAAS, the event loop was managed by the default twisted event loop. MAAS now uses an asyncio based event loop, driven by uvloop, which is targeted at improving internal performance.
+
+ - **Improved daemon management**
+   MAAS has changed the way daemons are run to allow users to see both ‘regiond’ and ‘rackd’ as processes in the process list.
```
Comment (caldav) [inline en/release-notes.md]: "`regiond` and `rackd`"  maybe?

File: en/release-notes.md [hunk 4]
```diff
@@ -25,6 +25,96 @@ The official Python client library for MAAS is available in the Ubuntu 18.04
...
+   As part of these changes, regiond workers are now managed by a master regiond process. In older versions of MAAS each worker was directly run by systemd. The master process is now in charge of ensuring workers are running at all times, and re-spawning new workers in case of failures. This also allows users to see the worker hierarchy in the process list.
+
+
+ - **Ability to increase the number of regiond workers**
+   Following the improved way MAAS daemons are run, further internal changes have been made to allow the number of regiond workers to be increased automatically. This allows MAAS to scale to handle more internal operations in larger environments.
+   While this capability is already available, it is not yet available by default. It will become available in the following milestone release.
```
Comment (caldav) [inline en/release-notes.md]: not yet available -> not yet enabled (?)

File: en/release-notes.md [hunk 5]
```diff
@@ -25,6 +25,96 @@ The official Python client library for MAAS is available in the Ubuntu 18.04
...
+
+
+- **Database query optimizations**
+   In the process of inspecting the internal operations of MAAS, it was discovered that multiple unnecessary database queries are performed for various operations. Optimising these requires internal improvements to reduce the footprint of these operations. Some areas that have been addressed in this release include:
+
+     *  When saving node objects (e.g. making any update of a machine, device, rack controller, etc), MAAS validated changes across various fields. This required an increased number of queries for fields, even when they were not being updated. MAAS now tracks specific fields that change and only performs queries for those fields.
+     Example: To update a power state, MAAS would perform 11 queries. After these improvements, , only 1 query is now performed.
```
Comment (caldav) [inline en/release-notes.md]: extraneous comma

File: en/release-notes.md [hunk 6]
```diff
@@ -25,6 +25,96 @@ The official Python client library for MAAS is available in the Ubuntu 18.04
...
+
+     * On every transaction, MAAS performed 2 queries to update the timestamp. This has now been consolidated into a single query per transaction.
+
+   These changes  greatly improve MAAS performance and database utilisation in larger environments. More improvements will continue to be made as we continue to examine various areas in MAAS.
+
+
+ - **UI optimisations**
+   MAAS is now being optimised to reduce the amount of data loaded in the websocket API to render the UI. This is targeted at only processing data for viewable information, improving various legacy areas. Currently, the work done in this area includes:
```
Comment (caldav) [inline en/release-notes.md]: This is targeted at only processing data -> The goal is to only process data

File: en/release-notes.md [hunk 7]
```diff
@@ -25,6 +25,96 @@ The official Python client library for MAAS is available in the Ubuntu 18.04
...
+
+#### Audit logging
+
+Continuing with the audit logging improvements, alpha2 now adds audit logging for all
+user actions that affect Hardware Testing & Commissioning.
+
+#### KVM pod improvements
+
+MAAS’ KVM pods was initially developed as a feature to help developers quickly iterate
+and test new functionality while developing MAAS. This, however, because a feature that
```
Comment (caldav) [inline en/release-notes.md]: because -> became

File: en/release-notes.md [hunk 8]
```diff
@@ -25,6 +25,96 @@ The official Python client library for MAAS is available in the Ubuntu 18.04
...
+allow not only developers, but also administrators to make better use of resources across
```
Comment (caldav) [inline en/release-notes.md]: allows

File: en/release-notes.md [hunk 9]
```diff
@@ -25,6 +25,96 @@ The official Python client library for MAAS is available in the Ubuntu 18.04
...
+their datacenter. Since the feature was initially create for developers, some features
+were lacking. As such, in 2.4 we are improving the usability of KVM pods:
+
+
+ - **Pod AZ’s**
+   MAAS now allows setting the physical zone for the pod. This helps administrators by conceptually placing their KVM pods in a AZ, which enables them to request/allocate machines on demand based on its AZ. All VM’s created from a pod will inherit the AZ.
```
Comment (caldav) [inline en/release-notes.md]: in a AZ -> in an AZ

File: en/release-notes.md [hunk 10]
```diff
@@ -25,6 +25,96 @@ The official Python client library for MAAS is available in the Ubuntu 18.04
...
+
+
+ - **Pod tagging**
+   MAAS now adds the ability to set tags for a pod. This allows administrators to use tags to allow/prevent the creation of a VM inside the pod using tags. For example, if the administrator would like a machine with a ‘tag’ named ‘virtual’, MAAS will filter all physical machines and only consider other VM’s or a KVM pod for machine allocation.
```
Comment (caldav) [inline en/release-notes.md]: "a tag named `virtual`" maybe?

File: en/release-notes.md [hunk 11]
```diff
@@ -25,6 +25,96 @@ The official Python client library for MAAS is available in the Ubuntu 18.04
...
+
+### Bug fixes
+
+Please refer to the following for all bug fixes in this release.
+
+https://launchpad.net/maas/+milestone/2.4.0alpha2
```
Comment (caldav) [inline en/release-notes.md]: Will this be rendered as a link automatically?

Comment (evilnick) [reply en/release-notes.md]: yes!

File: en/release-notes.md (hunk 1 above)
Comment (evilnick) [reply en/release-notes.md]: :+1:

File: en/release-notes.md (hunk 2 above)
Comment (evilnick) [reply en/release-notes.md]: :ok:

File: en/release-notes.md (hunk 3 above)
Comment (evilnick) [reply en/release-notes.md]: :+1:

File: en/release-notes.md (hunk 4 above)
Comment (evilnick) [reply en/release-notes.md]: :+1:

File: en/release-notes.md (hunk 9 above)
Comment (evilnick) [reply en/release-notes.md]: :ok:

File: en/release-notes.md (hunk 10 above)
Comment (evilnick) [reply en/release-notes.md]: :+1:

File: en/release-notes.md (hunk 8 above)
Comment (evilnick) [reply en/release-notes.md]: done

**Pull request merged**